### PR TITLE
Fix missing Open HMIS link for some users

### PR DESCRIPTION
**Pull request by gigxz**
## _Merging this PR_
- use the squash-merge strategy

## Description
This is a follow-up to https://github.com/greenriver/hmis-warehouse/pull/4745 which had the wrong logic. It was only showing "Open HMIS" link from the Warehouse if the user had any Access Control with a group that contained an entire Data Source. It _should_ show the link to a user that has access to ANY projects or orgs within the DS, which is what this change does.

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
Bug fix

## How to test
* impersonate a user that only has access to one project or org in hmis. they should see the Open HMIS link.

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- n/a If adding a new endpoint / exposing data in a new way, I have:
- [x] I have updated the documentation (or not applicable)
- [x] I have added spec tests (or not applicable)
- [x] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: drivers/hmis/app/models/hmis/group_viewable_entity.rb [hunk 1]
```diff
@@ -53,5 +53,14 @@ class GroupViewableEntity < GrdaWarehouseBase
...
       where(id: Array(entities).flat_map { |entity| includes_entity(entity).pluck(:id) })
     end
+
+    scope :includes_any_entity_in_data_source, ->(data_source) do
+      project_ids = data_source.projects.select(:id)
+      organization_ids = data_source.organizations.select(:id)
+
+      where(entity: data_source).
+        or(where(entity_type: Hmis::Hud::Project.sti_name, entity_id: project_ids)).
+        or(where(entity_type: Hmis::Hud::Organization.sti_name, entity_id: organization_ids))
```
Comment (eanders) [inline drivers/hmis/app/models/hmis/group_viewable_entity.rb:63]: I _think_ `GrdaWarehouse::ProjectCollectionMember` might make this simpler.  @ttoomey any chance you can take a quick look?

Comment (gigxz) [reply drivers/hmis/app/models/hmis/group_viewable_entity.rb:63]: I don't think we have the equivalent for HMIS, though it looks helpful. `GrdaWarehouse::ProjectCollectionMember` looks like it only has relationships to `Collection` which is the warehouse table

Comment (eanders) [reply drivers/hmis/app/models/hmis/group_viewable_entity.rb:63]: makes sense.

**Comment by gigxz** [comment]
Test failed on `rspec ./spec/models/curl_spec.rb:15 # Curl CA Certificate Works does throw an error when curling over https`, I haven't seen that one before

**Review by eanders — APPROVED** [review]

**Comment by eanders** [comment]
> Test failed on `rspec ./spec/models/curl_spec.rb:15 # Curl CA Certificate Works does throw an error when curling over https`, I haven't seen that one before

This is a test that was a canary for a bad certificate or bad implementation of CURL, sounds like it's been fixed, we should review, but not as part of this PR.

File: drivers/hmis/app/models/hmis/group_viewable_entity.rb (hunk 1 above)
Comment (ttoomey) [reply drivers/hmis/app/models/hmis/group_viewable_entity.rb:63]: Hmm, project collection member is collection_id, project_id so not directly applicable for this query. Similar shape though

**Pull request merged**